pkg/trace/agent/obfuscate.go

```diff
@@ -90,6 +93,31 @@ func (a *Agent) obfuscateSpan(span *pb.Span) {
 			return
 		}
 		span.Meta[tagElasticBody] = o.ObfuscateElasticSearchString(span.Meta[tagElasticBody])
+	case "system":
+		if span.Name == "command_execution" {
+			if !a.conf.Obfuscation.CommandExecution.Enabled {
+				return
+			}
+
+			v, ok := span.Meta[tagShellCommand]
+			if span.Meta == nil || !ok {
+				v, ok = span.Meta[tagExecCommand]
+				if span.Meta == nil || !ok {
+					return
+				}
+
+				cmdExec, cmdIndices, err := o.ObfuscateExecCommand(v)
+				if err != nil {
+					log.Debugf("Error obfuscating exec command %q: %v", v, err)
+					return
+				}
+
+				span.Meta[tagExecCommand] = cmdExec
+				span.Meta[tagShellIndices] = cmdIndices
+			} else {
+				span.Meta[tagShellCommand], span.Meta[tagShellIndices] = o.ObfuscateShellCommand(v)
+			}
```

Comment:
I think this logic could be simplified by inverting the cases. I'm also not sure why we would only obfuscate `tagExecCommand` if `tagShellCommand` is not present. Maybe there's not a case where tracers will produce a span with both, but it doesn't make sense to build that limitation into the system, especially when it complicates the logic.

I suggest something like:
```
if v, ok := span.Meta[tagShellCommand]; ok {
	span.Meta[tagShellCommand], span.Meta[tagShellIndices] = o.ObfuscateShellCommand(v)
}
if v, ok = span.Meta[tagExecCommand]; ok {
	cmdExec, cmdIndices, err := o.ObfuscateExecCommand(v)
	if err != nil {
		log.Debugf("Error obfuscating exec command %q: %v", v, err)
		// TODO: If we fail to obfuscate an exec command, is it safe to leave in the payload?
	} else {
		span.Meta[tagExecCommand] = cmdExec
		span.Meta[tagShellIndices] = cmdIndices
	}
}
```

Reply:
addressed in feb1189